### DISCUSSION TITLE
Output a normalised graphql file or introspection output file.

### DISCUSSION THREAD

**Discussion by ddskolla** [kobylynskyi/graphql-java-codegen#838] [Q&A]
Hi,

When a .grapql file in given as the input to the generator via the graphqlSchemas option, 
is there any way the generator is able to output a normalised version of that schema ?

The problem:
We use the kickstart/tools's @connection directive along with the relayConfig option to generate java related classes using your plugin.
We want to use the same .graphql file to generate our typescript clients/classes too, but because of the @connection annotation in the file, the typescript generator we use (https://www.graphql-code-generator.com) complains that it is an invalid .graphql file. 
So we were wondering if the generator could output a normalised version of the schema file after it parses the file and all the dependencies. 
In that case, we could use that normalised .graphql file as the source of truth for both the java and typescript generation.
Even if an introspection file could be spitted out by the generator (similar to what the typescript generator supports https://www.graphql-code-generator.com/docs/plugins/introspection), that would work.

Any help would be appreciated.

**Comment by kobylynskyi** [comment]
graphql-java-codegen uses graphql-java to validate and parse the schema, so if such feature exists, it should reside in graphql-java library.
Hope that helps.

**Reply by ddskolla** [reply]
thanks, looks like that is not possible at the moment on the gql-java end.
